src/figure_settings.py

```diff
@@ -128,9 +142,58 @@ def on_select(self, model, _pos, _n_items):
                 figure_settings.set_use_custom_style(True)
 
-    def set_axes_entries(self):
+    def set_axes_entries(self) -> None:
+        """
+        Sets the labels and visibility of all entries that are related to the
+        axes: scale, limits and label. Whenever two opposite axes are used
+        simultaniously, the title of the entry will get a directional prefix.
+        """
         visible_axes = self.get_application().get_data().get_used_positions()
-        for (direction, visible) in zip(_DIRECTIONS, visible_axes):
+        # Get the label for each direction, use directional prefix if
+        # two opposite X/Y axes are used simultaniously.
+        labels = {
+            "top": {
+                "min": _("Top X Axis Minimum") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Minimum"),
+                "max": _("Top X Axis Maximum") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Maximum"),
+                "scale": _("Top X Axis Scale") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Scale"),
+                "label": _("Top X Axis Label") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Label"),
+            },
+            "bottom": {
+                "min": _("Bottom X Axis Minimum") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Minimum"),
+                "max": _("Bottom X Axis Maximum") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Maximum"),
+                "scale": _("Bottom X Axis Scale") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Scale"),
+                "label": _("Bottom X Axis Label") if visible_axes[0]
+                and visible_axes[1] else _("X Axis Label"),
+            },
+            "left": {
+                "min": _("Left Y Axis Minimum") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Minimum"),
+                "max": _("Left Y Axis Maximum") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Maximum"),
+                "scale": _("Left Y Axis Scale") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Scale"),
+                "label": _("Left Y Axis Label") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Label"),
+            },
+            "right": {
+                "min": _("Right Y Axis Minimum") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Minimum"),
+                "max": _("Right Y Axis Maximum") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Maximum"),
+                "scale": _("Right Y Axis Scale") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Scale"),
+                "label": _("Right Y Axis Label") if visible_axes[2]
+                and visible_axes[3] else _("Y Axis Label"),
+            },
+        }
```

Comment:
There most definitely has to be a way to make this work as a set comprehension, as this is quite a bit of duplicated logic

Reply:
Yeah the thing is that the following strings should all be defined as a whole and not be broken up to avoid issues with translations. Not all languages has the adjectives in front of the word for example (see French):

Top X Axis Minimum
Top X Axis Maximum
Top X Axis Scale
Top X Axis Label
Bottom X Axis Minimum
Bottom X Axis Maximum
Bottom X Axis Scale
Bottom X Axis Label
X Axis Minimum
X Axis Maximum
X Axis Scale
X Axis Label
Left Y Axis Minimum
Left Y Axis Maximum
Left Y Axis Scale
Left Y Axis Label
Right Y Axis Minimum
Right Y Axis Maximum
Right Y Axis Scale
Right Y Axis Label
Y Axis Minimum
Y Axis Maximum
Y Axis Scale
Y Axis Label

That's a lot of strings that you can't just define upon runtime using a comprehension. So I haven't found a much neater way to do this myself so far.